### PR TITLE
pythonPackages.wordfreq and dependencies

### PR DESCRIPTION
**Pull request by ixxie**
Initialized the python wordfreq module at 2.0, with dependencies.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ixxie** [comment]
@dotlambda  - got time to review this?

File: pkgs/development/python-modules/langcodes/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    inherit pname version;
+    sha256 = "1axdiva2qglsjmnx2ak7i6hm0yhp6kbc4lcsgn8ckwy0nq1z3kr2";
+  };
+
+  propagatedBuildInputs = [ marisa-trie ];
+
+  disabled = pythonOlder "3.3";
+
+  meta = with lib; {
+    description = "langcodes is a toolkit for working with and comparing the standardized codes for languages, such as ‘en’ for English, ‘es’ for Spanish, and ‘zh-Hant’ for Traditional Chinese.";
```
Comment (dotlambda) [inline pkgs/development/python-modules/langcodes/default.nix]: https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

File: pkgs/development/python-modules/mecab-python3/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+buildPythonPackage rec {
+  pname = "mecab-python3";
+  version = "0.7";
+
+
+   src = fetchPypi {
+    inherit pname version;
+    sha256 = "007dg4f5fby2yl7cc44x6xwvcrf2w2ifmn0rmk56ss33mhs8l6qy";
+   };
```
Comment (dotlambda) [inline pkgs/development/python-modules/mecab-python3/default.nix]: indentation

File: pkgs/development/python-modules/mecab-python3/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+, buildPythonPackage
+, mecab
+, fetchPypi
+}:
+
+buildPythonPackage rec {
+  pname = "mecab-python3";
+  version = "0.7";
+
+
```
Comment (dotlambda) [inline pkgs/development/python-modules/mecab-python3/default.nix]: whitespace

File: pkgs/development/python-modules/wordfreq/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  version = "2.0";
+
+   src = fetchPypi {
+    inherit pname version;
+    sha256 = "0zbpjghv6z7xlq3445pb7fdbsz71iy4k2hv9lj67574lxmcgsabg";
+   };
+
+  # patch to relax version requirements for regex
+  # dependency to prevent break in upgrade
+  patches = [ ./relax-regex-dep.patch ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/wordfreq/default.nix]: I'd prefer using `substituteInPlace` in `postPatch`. Something like
```
postPatch = ''
  substituteInPlace setup.py \
    --replace "regex ==" "regex >="
'';
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Sorry for the late reply! I was on vacation.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.wordfreq python3.pkgs.wordfreq python3.pkgs.langcodes python2.pkgs.mecab-python3 python3.pkgs.mecab-python3 python2.pkgs.marisa-trie python3.pkgs.marisa-trie

**Comment by dotlambda** [comment]
> Ran 0 tests in 0.000s

Please make sure every package has an appropriate `checkPhase` if it provides tests. Otherwise, please set `doCheck = false` and add an inline comment with the reason.

**Comment by ixxie** [comment]
@dotlambda I think I managed to address all your concerns but one: I tried to get wordfreq's tests going by switching to `fetchFromGitHub` and triggering nose tests like you instructed me to do for `hdbscan`. However the tests fail due to a missing `mecab-ko-dic` dependency, and [the first hit on Google](https://bitbucket.org/eunjeon/mecab-ko-dic) is all in Korean, which makes sense for a Korean dictionary package. I am not sure how to proceed at the moment, perhaps you have an idea?

**Comment by dotlambda** [comment]
I updated all expressions so that the tests pass and so that all packages are tested. Please have a look.

**Comment by ixxie** [comment]
Awesome, so I guess we are done :)

**Pull request merged**